contract-ts/03-owner-claims-winner-gets-nft/src/contract.ts

```diff
@@ -65,18 +65,10 @@ class AuctionContract {
     assert(!this.claimed, "Auction has been claimed");
     this.claimed = true;
 
-    // Returning joint promise is currently prohibited.
-    return NearPromise.new(this.nft_contract)
-      .functionCall("nft_transfer", JSON.stringify({ receiver_id: this.highest_bid.bidder, token_id: this.token_id }), BigInt(1), TWENTY_TGAS)
-      .then(
-        NearPromise.new(this.auctioneer).transfer(this.highest_bid.bid)
-      )
-      .asReturn()
-
-  }
-
-  @call({ privateFunction: true })
-  nft_transfer_callback({ }): BigInt {
-    return;
+    if (this.highest_bid.bidder != near.currentAccountId()) {
```

Comment:
Many NFT and FT contracts have a check that you cannot transfer to yourself including the [NFT contract we are using](https://github.com/near-examples/nft-tutorial/blob/main/nft-contract-approval/src/internal.rs#L169). Is it still ok because this is a cross contract call and if it fails who cares?

Reply:
Indeed